src/main/java/org/checkerframework/checker/builder/TypesafeBuilderAnnotatedTypeFactory.java

```diff
@@ -158,38 +165,100 @@ public Void visitExecutable(AnnotatedTypeMirror.AnnotatedExecutableType t, Void
+        builderKind = BuilderKind.AUTO_VALUE;
+        assert hasAnnotation(nextEnclosingClass, AutoValue.class)
+            : "class " + nextEnclosingClass.getSimpleName() + " is missing @AutoValue annotation";
 
-        assert hasAnnotation(autoValueClass, AutoValue.class)
-            : "class " + autoValueClass.getSimpleName() + " is missing @AutoValue annotation";
+      } else if (hasAnnotation(enclosingClass, lombok.Generated.class)
+          && enclosingClass.getSimpleName().toString().endsWith("Builder")) {
+        builderKind = BuilderKind.LOMBOK;
+      }
+
+      if (builderKind != BuilderKind.NONE) {
+        // get the name of the method
+        String methodName = methodTree.getName().toString();
 
         if ("build".equals(methodName)) {
           // determine the required properties and add a corresponding @CalledMethods annotation
-          List<String> requiredProperties = getRequiredProperties(autoValueClass);
+          List<String> requiredProperties = getRequiredProperties(nextEnclosingClass, builderKind);
           AnnotationMirror newCalledMethodsAnno =
-              createCalledMethodsForProperties(requiredProperties);
+              createCalledMethodsForProperties(requiredProperties, builderKind);
           t.getReceiverType().addAnnotation(newCalledMethodsAnno);
         }
       }
 
       return super.visitExecutable(t, p);
     }
 
+    /**
+     * computes the required properties of a builder class
+     *
+     * @param builderClass the class whose builder is to be checked
+     * @param builderKind the framework by which the builder will be generated
+     * @return a list of required property names
+     */
+    private List<String> getRequiredProperties(
+        final ClassTree builderClass, final BuilderKind builderKind) {
+      switch (builderKind) {
+        case AUTO_VALUE:
+          return getAutoValueRequiredProperties(builderClass);
+        case LOMBOK:
+          return getLombokRequiredProperties(builderClass);
+        default:
+          return Collections.emptyList();
+      }
+    }
+
+    /**
+     * computes the required properties of a @lombok.Builder class, i.e., the names of the fields
+     * with @lombok.NonNull annotations
+     *
+     * @param lombokClass the class with the @lombok.Builder annotation
+     * @return a list of required property names
+     */
+    private List<String> getLombokRequiredProperties(final ClassTree lombokClass) {
+      List<String> requiredPropertyNames = new ArrayList<>();
+      List<String> defaultedPropertyNames = new ArrayList<>();
+      for (Tree member : lombokClass.getMembers()) {
+        if (member.getKind() == Tree.Kind.VARIABLE) {
+          VariableTree fieldTree = (VariableTree) member;
+          for (AnnotationTree atree : fieldTree.getModifiers().getAnnotations()) {
+            AnnotationMirror anm = TreeUtils.annotationFromAnnotationTree(atree);
+            if (AnnotationUtils.areSameByClass(anm, lombok.NonNull.class)) {
```

Comment:
Are we sure that Lombok checks for `@lombok.NonNull` and not any annotation named `@NonNull`?

Reply:
Good idea. I've changed the check so that it's consistent with the list of annotations that Lombok uses.